variable/test/comparison_test.cpp

```diff
@@ -70,3 +70,105 @@ TEST(ComparisonTest, tolerance_type_mismatch) {
   EXPECT_THROW(is_approx(a, b, 0 /*implicit int*/), except::TypeError);
   EXPECT_NO_THROW(is_approx(a, b, 0.0f));
 }
+
+TEST(ComparisonTest, less_variances_test) {
+  const auto a = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0},
+                                     Variances{1.0, 1.0});
+  const auto b = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{0.0, 3.0},
+                                     Variances{1.0, 1.0});
+  EXPECT_THROW(less(a, b), std::runtime_error);
+}
+
+TEST(ComparisonTest, less_dtypes_test) {
+  const auto a = makeVariable<float>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0});
+  const auto b = makeVariable<int>(Dims{Dim::X}, Shape{2}, Values{0, 1});
+  EXPECT_THROW(less(a, b), std::runtime_error);
+}
+
+TEST(ComparisonTest, less_units_test) {
+  const auto a = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{1.0, 2.0});
+  auto b = makeVariable<double>(Dims{Dim::X}, Shape{2}, Values{0.0, 3.0});
+  b.setUnit(units::m);
+  EXPECT_THROW(less(a, b), std::runtime_error);
+}
+
+template <typename T> class LessTest : public ::testing::Test {};
+template <typename T> class GreaterTest : public ::testing::Test {};
+template <typename T> class LessEqualTest : public ::testing::Test {};
+template <typename T> class GreaterEqualTest : public ::testing::Test {};
+template <typename T> class EqualTest : public ::testing::Test {};
+template <typename T> class NotEqualTest : public ::testing::Test {};
+
+using CompareTestTypes = ::testing::Types<double, float, int64_t, int32_t>;
+TYPED_TEST_SUITE(GreaterTest, CompareTestTypes);
+TYPED_TEST_SUITE(LessTest, CompareTestTypes);
+TYPED_TEST_SUITE(LessEqualTest, CompareTestTypes);
+TYPED_TEST_SUITE(GreaterEqualTest, CompareTestTypes);
+TYPED_TEST_SUITE(EqualTest, CompareTestTypes);
+TYPED_TEST_SUITE(NotEqualTest, CompareTestTypes);
```

Comment:
Do we need typed tests at all on this level? You have already tested that this works in `element_comparison_test.cpp`.